### PR TITLE
Problem: base fee is adjusting immediately after upgrade while London hardfork not enabled

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #333
Solution:
- disable feemarket in upgrade handler
- update integration tests

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -697,7 +698,19 @@ func New(
...
 		// The default genesis parameters of `feemarket` module are fine, because the `InitialBaseFee (1000000000)` is much lower than the current minimal gas price,
 		// so it don't have effect at the beginning, we can always adjust the parameters through the governance later.
-		return app.mm.RunMigrations(ctx, app.configurator, fromVM)
+		updatedVM, err := app.mm.RunMigrations(ctx, app.configurator, fromVM)
+		if err != nil {
+			return nil, err
+		}
+		// Override feemarket parameters
+		app.FeeMarketKeeper.SetParams(ctx, feemarkettypes.Params{
+			NoBaseFee:                true, // enable later through gov proposal
```
Comment (thomas-nguy) [inline app/app.go:707]: another solution couldnt be to set `london_block` at the upgrade height?

Comment (yihuang) [reply app/app.go:707]: do you mean change parameter in upgrade handler, or open another parameter change gov proposal?

Comment (thomas-nguy) [reply app/app.go:707]: change parameter in upgrade handler. 

If I understood correctly, the problem is that the base fee is activated while the london hardfork is not?

Comment (yihuang) [reply app/app.go:707]: we might need to set the fee market parameters together too?

Comment (thomas-nguy) [reply app/app.go:707]: yes we will need to update the handler with the correct feemarket parameters

File: app/app.go [hunk 2]
```diff
@@ -697,7 +698,19 @@ func New(
 	app.UpgradeKeeper.SetUpgradeHandler(plan0_7_0, func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
```
Comment (thomas-nguy) [inline app/app.go:698]: lets create another upgrade plan?

Comment (yihuang) [reply app/app.go:698]: I converted to draft for now, until we made a decision on this issue.

Comment (yihuang) [reply app/app.go:702]: Actually, I think it's fine because it'll be released in different versions of binaries.

**Comment by yihuang** [comment]
After a second thought, I think this PR is good to merge, we can always change our mind if we want to hard code the feemarket parameters in upgrade handler later.

**Comment by yihuang** [comment]
wait for final feemarket parameters to hardcode in upgrade handler.

**Comment by yihuang** [comment]
will hardcode feemarket parameters in upgrade handler.

**Pull request closed without merging**